libdnf5/base/base.cpp

```diff
@@ -173,6 +173,11 @@ void Base::setup() {
         config.get_logdir_option().set(Option::Priority::INSTALLROOT, full_path);
     }
 
+    // If only cached metadata are used, disable rebuilding solv cache files.
+    if (config.get_cacheonly_option().get_value() != "none") {
+        config.get_build_cache_option().set(Option::Priority::RUNTIME, false);
```

Comment:
> Can I ask what is the benefit or reason behind this?

Oh, I remember, this occurred when I started implementing this, and the user cache was initially pointing to the root's cache. As a result, users didn't have the necessary permissions when the solv files attempted to rebuild in the root cache.

> Do the solv cache files somehow get rebuild unnecessarily?

So, yes, I believe there were, and probably still are, some use cases where the cache files are regenerated even in cacheonly mode. I haven't thoroughly analyzed whether this is "unnecessary".

> I am thinking about a scenario where a user has only metadata in cache and wants to run in cacheonly. They won't be able to take advantage of the solv files speed up.

Hmm, I may not fully understand. Are there scenarios where we have existing metadata in the cache, run in cacheonly mode, and still intend to rebuild solv files?

Reply:
But considering the current implementation of cloning the root cache, there is a possibility of having metadata present but solv files unavailable. Therefore, I'll drop this commit. It would, in any case, only represent a potential optimization in certain use cases that are not even clear if they can occur.